pkg/kapp/app/recorded_app.go

```diff
@@ -217,7 +222,12 @@ func (a *RecordedApp) labeledApp() (*LabeledApp, error) {
 		return nil, err
 	}
 
-	sel := labels.Set(meta.Labels()).AsSelector()
+	appLabels := meta.Labels()
+	for key, val := range a.labels {
+		appLabels[key] = val
+	}
+
+	sel := labels.Set(appLabels).AsSelector()
```

Comment:
we should not be adding labels used to label app to LabeledApp

Reply:
ok yeah, make sense to me. But now I'm not getting if I don't set them how I can get additional labels other than `kapp.k14s.io/app=<something>` with the help of label selector [here](https://github.com/vmware-tanzu/carvel-kapp/blob/edeb9dd9a923e15187f88999e983cb5762a14faa/pkg/kapp/cmd/app/list.go#L100)

Will it be a good idea to have a separate `getLabels` method for App interface and then concatenate this into what we are getting from `sel.String()` ?

Reply:
Make sense. As this functionality was added additionally will it good idea to raise a separate issue for showing all the labels while doing `kapp ls --column name,label` and let's move this out of the scope of this PR ?

I'm removing this change from this PR as of now.